docker-compose.yml

```diff
@@ -0,0 +1,68 @@
+version: '2.1'
+volumes:
+ oauth2-mysql-data:
+ hospital-service-data:
+
+services:
+  oauth2:
+    build: oauth-authorization-server
+    environment:
+    - OAUTH2_DB_USER=root
+    - OAUTH2_DB_PASS=secret
+    - OAUTH2_DB_URL=oauth2-mysql
+    - OAUTH2_DB_PORT=3306
+    command: ["--spring.profiles.active=prod"]
+    ports:
+    - "8090:8090"
+    depends_on:
+      oauth2-mysql:
+        condition: service_healthy
+
+  oauth2-mysql:
+    image: mysql:5.7
+    container_name: oauth2-mysql
+    volumes:
+      - oauth2-mysql-data:/var/lib/mysql
+    restart: always
+    environment:
+      - MYSQL_DATABASE=oauth2
+      - MYSQL_ROOT_USER=root
+      - MYSQL_ROOT_PASSWORD=secret
+    ports:
+      - "33061:3306"
+    healthcheck:
+      test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
+      timeout: 20s
+      retries: 10
+  
+  hospital:
+    build: hospital-service
+    environment:
+      - HOSPITAL_DB_USERNAME=root
+      - HOSPITAL_DB_PASSWORD=admin
+      - HOSPITAL_DB_URL=hospital-mysql
+      - HOSPITAL_DB_PORT=3306
+    command: ["--spring.profiles.active=prod"]
+    ports:
+      - "8000:8000"
+    depends_on:
+      hospital-mysql:
+       condition: service_healthy
+  
+  hospital-mysql:
+    image: mysql:5.7
+    container_name: hospital-mysql
+    volumes:
+      - hospital-service-data:/var/lib/mysql
+    restart: always
+    environment:
+      - MYSQL_DATABASE=hospital-mysql
+      - MYSQL_ROOT_USER=root
+      - MYSQL_ROOT_PASSWORD=admin
+    ports:
+      - "3306:3306"
```

Comment:
1. For me 3306:3306 didn't work, I had to do 33061:3306.
Would you please try with 33061, if it works leave it like that? I've seen many posts of people mapping 33061 and not 3306.

2. Have you tried starting both oauth and hospital services with prod profile?
e.g. docker compose up; ?

Hospital database is using the same port as oauth database, this doesn't look like it would work.
Please change hospital database to another port.

Reply:
Yes, it does not work. I have changed it in my latest commit.